services/src/unit-test/java/uk/gov/moj/sdt/services/messaging/IndividualRequestMdbTest.java

```diff
@@ -73,76 +56,80 @@ public void setUp() {
-
+        assertTrue(true,TEST_SUCCESSFULLY_COMPLETED);
     }
 
     /**
      * Test method to test the scenario where the service throws a Data Access Exception.
      *
      * @throws JMSException if any during the test setup
      */
+
     @Test
-    public void readMessageDataAccessException() throws JMSException {
+    public void readMessageDataAccessExceptionTest() throws JMSException {
         individualRequestMdb = new IndividualRequestMdb(mockTargetSubmissionService);
 
-        // Create the actual message to send.
+        // Create the actual message to send
         final ISdtMessage sdtMessage = new SdtMessage();
         sdtMessage.setSdtRequestReference("ReadTest1");
 
         // Wrap the message in the ObjectMessage
-        final ObjectMessage objMessage = EasyMock.createMock(ObjectMessage.class);
-        objMessage.setObject((Serializable) sdtMessage);
-        EasyMock.expectLastCall().anyTimes();
-
-        EasyMock.expect(objMessage.getObject()).andReturn((Serializable) sdtMessage);
-        EasyMock.replay(objMessage);
+        final ObjectMessage objMessage = mock(ObjectMessage.class);
+        when(objMessage.getObject()).thenReturn((Serializable) sdtMessage);
 
-        mockTargetSubmissionService.processRequestToSubmit(sdtMessage.getSdtRequestReference());
-
-        EasyMock.expectLastCall().andThrow(new RuntimeException("Data error occurred"));
-
-        EasyMock.replay(mockTargetSubmissionService);
+        doThrow(new RuntimeException("Data error occurred"))
+            .when(mockTargetSubmissionService).processRequestToSubmit(sdtMessage.getSdtRequestReference());
 
         try {
             individualRequestMdb.readMessage(objMessage);
-            Assert.fail("RuntimeException was expected to be thrown");
+            fail("RuntimeException was expected to be thrown");
         } catch (final RuntimeException re) {
-            Assert.assertTrue("Expected to throw the exception", true);
+            assertTrue(true,"Expected to throw the exception");
         }
 
-        EasyMock.verify(mockTargetSubmissionService);
-        EasyMock.verify(objMessage);
+        Mockito.verify(mockTargetSubmissionService).processRequestToSubmit(sdtMessage.getSdtRequestReference());
+        Mockito.verify(objMessage).getObject();
+
+        assertTrue(true,TEST_SUCCESSFULLY_COMPLETED);
+    }
+
+
+    @Test
+    public void readMessageJMSExceptionTest() throws JMSException {
+        individualRequestMdb = new IndividualRequestMdb(mockTargetSubmissionService);
 
-        Assert.assertTrue("Test successfully completed", true);
+        // Wrap the message in the ObjectMessage
+        final ObjectMessage objMessageMock = mock(ObjectMessage.class);
+
+        doThrow(new JMSException("JMS error occurred")).when(objMessageMock).getObject();
+
+        try {
+            individualRequestMdb.readMessage(objMessageMock);
+            fail("Exception should be thrown");
+        } catch (final Exception e) {
```

Comment:
Change exception type to JMSException as that's the exception you're expecting to be thrown.

Reply:
When I do this I get errors in the code due to the method then throwning runtime error.  So I've cause the exception and done the foolwing :- assertEquals(e.getCause().getMessage(),"JMS error occurred");

Reply:
You're right - the method being tested catches the JMSException and throws a RuntimeException in it's place.  In that case change the catch block to catch a RuntimeException rather than an Exception as that's the type of exception that's expected.

Might as well remove the assertTrue() from the catch block as the assertEquals() gives us the confidence that the right exception has been thrown.

Reply:
Done

Reply:
It's still there!